### PR TITLE
fix(star-adventurer-gti): correct raw-vs-folded encoder handling in dec helpers (#242)

### PR DESCRIPTION
**Pull request by ivonnyssen**
## Summary

Resolves a latent correctness bug discovered while investigating issue #242: `flip_slew_dec_delta` and `side_of_pier` operate on raw encoder counters but apply a half-classification (`|.| > cpr_dec/4`) that is only meaningful on the folded canonical-band value. When raw drifts into the disagreement zone `(3·cpr/4, 5·cpr/4)` — possible after through-wrap flip slews, manual `:E` writes, or power-up encoder noise — the heuristic flips the verdict relative to the OTA's physical position, with downstream effects on routing safety and ASCOM-client-visible pier-side reporting.

The originally-filed bug in #242 (a sign-blindness pattern in `flip_slew_dec_delta` analogous to the RA fix in `072cc72`) turned out *not* to exist: after enumerating every realistic flip-slew (current, target) subcase the heuristic is correct for in-band raw values. But the analysis surfaced a real latent issue (raw vs folded) plus the case for porting the path-aware structure from RA for symmetry and robustness.

## Changes

Three commits, each independently meaningful:

1. **`7aefcf3`** — fold raw `current_ticks` in `flip_slew_dec_delta` before the half-classification. Minimal targeted fix for the latent bug.
2. **`3cd4005`** — same fix applied to `side_of_pier`. Promotes `fold_delta_to_canonical` from `mount_device.rs` to `coordinates.rs` (where the rest of the axis-arithmetic lives) so both call sites import the same helper. This is the more impactful of the two folds — `side_of_pier` is user-visible through the ASCOM `SideOfPier` / `DestinationSideOfPier` getters and feeds the `is_flip_slew` decision plus `set_side_of_pier`'s no-op check.
3. **`3cba41a`** — path-aware rewrite of `flip_slew_dec_delta` to mirror `flip_slew_ra_delta`'s structure. Replaces the heuristic with a direct check of whether the linear encoder sweep `[current, current + canonical]` crosses any modular replica of the below-horizon pole. Promotes the test-module `path_crosses_pole` helper to production `canonical_path_crosses_pole`. Renders commit 1's fold redundant (the path-aware version is naturally robust to raw outside the canonical band).

## Test plan

- [x] `cargo rail run --profile commit -q` — 298 tests pass (was 297 pre-merge; +1 from main's `home_pose` change)
- [x] `cargo fmt --check` clean
- [x] New regression tests:
  - `coordinates::tests::side_of_pier_folds_raw_outside_canonical_band` — covers positive + negative disagreement zone, both hemispheres
  - `mount_device::tests::flip_slew_dec_delta_handles_raw_current_outside_canonical_band` — raw in disagreement zone, sanity-checks path-aware preserves the safe canonical
- [x] All 8 pre-existing `flip_slew_dec_delta_*` semantic tests pass unchanged (verifies behaviour identity for in-band inputs)
- [x] All 4 `fold_delta_to_canonical_*` tests still pass (now in `coordinates` module alongside the function)
- [ ] Hardware re-verification deferred to next session — purely defensive changes, no in-band behaviour change

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: services/star-adventurer-gti/src/mount_device.rs [hunk 1]
```diff
@@ -908,19 +879,39 @@ fn flip_slew_dec_delta(
...
+    // `|delta| ≤ cpr` by the canonical-fold invariant, but `start` can
+    // sit far outside `[-cpr/2, +cpr/2)` after through-wrap flips —
+    // scanning `k ∈ [-2, +2]` covers every replica that could fall in
+    // `[lo, hi]`.
+    (-2..=2).any(|k| {
+        let p = pole_ticks + k * cpr_i;
+        p >= lo && p <= hi
+    })
+}
+
```
[the remark this answers is not in the document]
Comment (ivonnyssen) [reply services/star-adventurer-gti/src/mount_device.rs]: Addressed in fc4b4d2 — replaced the fixed `k ∈ -2..=2` scan with a direct interval-contains-multiple check via `div_euclid`. Operates on any `start` within i32 range with no replica-index assumptions. Added a regression test that pins the k=±3 wire-boundary case for both hemispheres and exercises the end-to-end `flip_slew_dec_delta` behaviour (canonical positive at +8.3M raw correctly flipped to the long way).

File: services/star-adventurer-gti/src/coordinates.rs [hunk 2]
```diff
@@ -512,6 +519,34 @@ fn fold_to_signed(value: f64, period: f64) -> f64 {
...
+/// the counter can run from `−2²³` to `+2²³ − 1` before the codec's
+/// 24-bit field wraps. A through-wrap meridian-flip slew can therefore
+/// leave the encoder counter outside the canonical `[−cpr/2, +cpr/2)`
+/// band (e.g. `−1.89M` for a flip that landed physically at `+11.5 h`,
+/// modular `+1.74M`). Without folding, the next slew's
+/// `target_ticks − current_ticks` would order a full extra revolution,
+/// and the [`side_of_pier`] half-classification would misread the
+/// physical position. This helper collapses the raw value to its
+/// canonical-band equivalent.
+pub fn fold_delta_to_canonical(value: i32, cpr: u32) -> i32 {
```
[the remark this answers is not in the document]
Comment (ivonnyssen) [reply services/star-adventurer-gti/src/coordinates.rs]: Agreed. The function is now used for both folding deltas (slew planning, pickup loop) and folding raw positions (`side_of_pier`), and the "delta" in the name is misleading. The actual operation is just "fold a signed value into the canonical band `[-cpr/2, +cpr/2)`" — domain-neutral.

Will rename to `fold_to_canonical_band(value, cpr)` (the canonical band is already a well-defined concept in the codebase, used in slew planning + the past-the-pole encoder analysis). Updates call sites in `mount_device.rs` (4), `coordinates.rs` (the `side_of_pier` call), and the four `fold_delta_to_canonical_*` test names. Keeping it as a single rename rather than aliasing — two names for the same operation would invite divergence later.

File: services/star-adventurer-gti/src/mount_device.rs [hunk 3]
```diff
@@ -5822,6 +5754,87 @@ mod tests {
...
+        // is k=-2: -907_200 - 2·3_628_800 = -8,164,800, which IS in
+        // [-10M, -8.3M] — well within the old `-2..=2` window. The
+        // negative wire boundary doesn't suffer the same k=±3 miss
+        // because the pole is at -cpr/4, not +cpr/4, so the asymmetry
+        // shifts the danger band onto the positive side only for
+        // northern observers. Southern observers (pole at +cpr/4) get
+        // the mirror — exercise that too.
+        let pole_south = cpr_i / 4;
+        // Sweep [-10M, -8.3M] contains the k=-3 replica at
+        // +907_200 - 3·3_628_800 = -9_979_200.
```
[the remark this answers is not in the document]
Comment (ivonnyssen) [reply services/star-adventurer-gti/src/mount_device.rs]: Agreed. That comment block is a record of me working through the geometry while writing the test, complete with an arithmetic backtrack — useful while I was figuring it out, noise once the test settled. Will split into two focused tests: one for the k=+3 north case (helper-direct + end-to-end through `flip_slew_dec_delta` with `northern=true`), one for the k=-3 south case (same shape, `northern=false`). Each gets a tight derivation comment showing just the modular arithmetic for that hemisphere.

Bonus: splitting also gives the south case its own end-to-end assertion through `flip_slew_dec_delta`, which the original test was missing — it only exercised the helper directly for south.
